config: add secret path env var

happysalada opened the pull request:
hey, I had a go at #132 
this is just an early attempt.
I've added a flake.nix which gives a development environment for anyone wanting to contribute. Happy to remove it in case you don't like it.
I've added what I thought would be a solution for reading secrets from path.
I haven't optimize the allocation, this should probably be a Cow, but I wanted to see how you felt about this.
Note that there is a bunch of clippy warnings that I didn't touch.
Let me know if anything, I'm flexible.

s3rius commented on src/config.rs:
```diff
@@ -62,12 +62,25 @@ pub struct StorageOptions {
+    /// S3 access key path.
+    ///
+    /// This parameter is used fo s3-based storages.
+    /// It's a version of s3_access_key more suitable for secrets.
```
```suggestion
    /// It's a path to file that has s3-access-key inside.
```

happysalada commented:
I've removed the flake.nix. However this is not just a nixos feature. I'm on macos, you just need the nix package manager to benefit from it.
This sets up a development environment with a precise rust version, openssl version... Those mostly work out of the box, but sometimes they don't, it's just a way to make sure everyone has the same environment no matter what is installed on their machine.
Don't worry about it though, I've removed it, it's simpler that way.

s3rius commented on src/storages/models/available_stores.rs:
```diff
@@ -54,3 +67,21 @@ impl AvailableStores {
+
+// TODO this should probably be a COW
+fn from_string_or_path(variable: &Option<String>, path: &Option<PathBuf>) -> Option<String> {
+    if let Some(variable) = variable {
```
Maybe rewrite it wit match statement?

s3rius commented on src/storages/models/available_stores.rs:
```diff
@@ -54,3 +67,21 @@ impl AvailableStores {
 }
+
+// TODO this should probably be a COW
+fn from_string_or_path(variable: &Option<String>, path: &Option<PathBuf>) -> Option<String> {
```
As I see, it doesn't return Option, it returns string. Because otherwise it panics. 

Maybe you can replace Option with result or make it return just string.

s3rius reviewed (changes requested):
Everything with docs look nice! Thank you. I just took a closer look to `from_string_or_path` function. It looks good and I'm pretty sure it works, but I guess it may be even better.

happysalada commented:
@s3rius I agree with all of your comments except maybe the match.
The match with 2 branches I feel is less readable than the if else.
Let me know if anything of course.

s3rius commented:
Okay. Agree, that match can be not as readable as if else here.

happysalada commented:
There was a clippy lint failing, I've just force pushed to fix it.
